src/Core/Field.cpp

```diff
@@ -336,6 +355,183 @@ void writeText(const Tuple & x, WriteBuffer & buf)
     writeFieldText(DB::Field(x), buf);
 }
 
+void readBinary(Map & x, ReadBuffer & buf)
+{
+    size_t size;
+    DB::readBinary(size, buf);
+
+    for (size_t index = 0; index < size; ++index)
+    {
+        UInt8 type;
+        DB::readBinary(type, buf);
+
+        switch (type)
```

Comment:
Consider using `Field::dispatch` + overloaded functions, to avoid enumerating all Field types in many places.

Reply:
As an example, see `writeFieldText` or `toString(Field)` in `Field.cpp`.